src/Functions/reinterpretAs.cpp

```diff
@@ -25,30 +26,60 @@ namespace ErrorCodes
 
 namespace
 {
-
-template <typename ToDataType, typename Name>
-class FunctionReinterpretStringAs : public IFunction
+template <typename ToDataType, typename Name, bool support_between_float_integer>
+class FunctionReinterpretAs : public IFunction
 {
+    template <typename F>
+    static bool castType(const IDataType * type, F && f)
+    {
+        return castTypeToEither<DataTypeUInt32, DataTypeInt32, DataTypeUInt64, DataTypeInt64, DataTypeFloat32, DataTypeFloat64>(
+            type, std::forward<F>(f));
+    }
+
+    template <typename From, typename To>
+    static void reinterpretImpl(const PaddedPODArray<From> & from, PaddedPODArray<To> & to)
+    {
+        const auto * from_reinterpret = reinterpret_cast<To *>(const_cast<From *>(from.data()));
```

Comment:
This is slightly incorrect due to "strict aliasing".
It's better to utilize `unalignedLoad` function.

Reply:
ok!